middleware/kubernetes/autopath.go

```diff
@@ -3,30 +3,46 @@ package kubernetes
 import (
 	"fmt"
 
+	"github.com/coredns/coredns/middleware"
 	"github.com/coredns/coredns/request"
 
 	"k8s.io/client-go/1.5/pkg/api"
 )
 
+// AutoPath implements the AutoPathFunc call from the autopath middleware.
+// It returns a per-query search path or an error indicating no searchpathing should happen.
 func (k *Kubernetes) AutoPath(state request.Request) ([]string, error) {
+	// Check if the query falls in a zone we are actually authoriative for and thus if we want autopath.
+	zone := middleware.Zones(k.Zones).Matches(state.Name())
+	if zone == "" {
+		return nil, fmt.Errorf("kubernetes: no authoriative for %s", state.Name())
+	}
+
 	ip := state.IP()
 
-	pod := k.PodWithIP(ip)
+	pod := k.podWithIP(ip)
 	if pod == nil {
-		return nil, fmt.Errorf("kubernets: no pod found for %s", ip)
+		return nil, fmt.Errorf("kubernetes: no pod found for %s", ip)
```

Comment:
Do we need this to be an error? I think it should just mean "do not attempt the auto path optimization" but not be considered an error condition, per se